teps/0050-ignore-task-failures.md

```diff
@@ -0,0 +1,190 @@
+
+
+### Goals
+
+* Design a task failure strategy so that the pipeline author can control the behavior of the underlying tasks 
+  and decide whether to continue executing the rest of the graph in the event of failure.
+
+* Be applicable to any pipeline with references to the tasks in a catalog or inlined task specifications.
+
+### Non-Goals
+
+* Not an alternative to combining the tasks in a pipeline which is covered in
+  [TEP-0044 Composing Tasks with Tasks](https://github.com/tektoncd/community/pull/316).
+
+* Not optimizing pipeline runtime which is covered in
+  [TEP-0046 PipelineRun in a Pod](https://github.com/tektoncd/community/pull/318).
+
+## Requirements
+
+* Users should be able to use any task from the catalog without having to alter its specification to allow that task to
+  fail without stopping the execution of a pipeline.
+
+* It should be possible to know that a task failed, and the rest of the graph was allowed to continue by observing
+  the status of the `PipelineRun`.
+
+* Do not fail the pipeline if any of the tasks fail and continue executing rest of the tasks.
+
+### Use Cases
+
+* As a pipeline author, I would like to design a pipeline where a task running
+  [unit tests](https://github.com/tektoncd/catalog/tree/master/task/golang-test/0.1) might fail,
+  but the pipeline can continue running integration tests and deploying an application to a staging cluster, so that the
+  application can be shared with other developers for early feedback.
+
+  For example, do not fail the pipeline if `unit tests` fail. Continue deploying to a staging cluster if integration tests
+  succeed.
+
+  ```
+           |                     |
+           v                     v
+       unit tests      integration tests
+                                |
+                                v
+                      deploy to staging cluster
+  ```
+
+* As a pipeline author, I would like to design a pipeline where a task running
+  [linting](https://github.com/tektoncd/catalog/tree/master/task/golangci-lint/0.1) might fail,
+  but can continue running tests, so that my pipeline can report failures from the linting and all the tests.
+
+  For example, do not fail the pipeline if `linting` fail, continue reporting the linter analysis.
+
+  ```
+           |                     |
+           v                     v
+       linting               unit tests
+           |                     |
+           v                     v
+   report linter output    integration tests
+  ```
+
+  In this example, `linting` and `unit tests` are executed in parallel. This specific use case can be
+  supported by `pipeline in pipeline` approach by creating two separate pipelines,
+  (1) `linting` -> `report linter output` and (2) `unit tests` -> `integration tests`.
+  `pipeline in pipeline` approach would be better fit when you want `linting` to fail the pipeline.
```

Comment:
is `the pipeline` here referring to the first sub-pipeline or the whole pipeline?

if pipelines in pipelines approach is used, it might be better for `report linter output` to be a finally task which would then report the linting output and the tests sub-pipeline would execute independently in parallel so `linting` failure won't affect it

what do you think?

Reply:
> is the pipeline here referring to the first sub-pipeline or the whole pipeline?

Its referring to the sub pipeline and in turn the whole pipeline since the sub pipeline status would be reflected in the overall pipelineRun status.

> what do you think?

Yup, having `report linter output` under `finally` won't be affected in terms of the execution by the `linting` failure but that sub pipeline would still result in `failure`. 

Here the idea is to treat `linting` as a warning. `linting` is part of the pipeline but its failure has no impact on the execution of the rest of the pipeline or the overall status.